test/integration/delete_test.go

```diff
@@ -0,0 +1,114 @@
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/config"
+	"github.com/content-services/content-sources-backend/pkg/dao"
+	"github.com/content-services/content-sources-backend/pkg/db"
+	m "github.com/content-services/content-sources-backend/pkg/instrumentation"
+	"github.com/content-services/content-sources-backend/pkg/tasks"
+	"github.com/content-services/content-sources-backend/pkg/tasks/client"
+	"github.com/content-services/content-sources-backend/pkg/tasks/queue"
+	"github.com/content-services/content-sources-backend/pkg/tasks/worker"
+	uuid2 "github.com/google/uuid"
+	"github.com/openlyinc/pointy"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/zerolog/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+// This is a delete integration tests without any snapshotting
+type DeleteTest struct {
+	Suite
+	dao   *dao.DaoRegistry
+	queue queue.PgQueue
+}
+
+func (s *DeleteTest) SetupTest() {
+	s.Suite.SetupTest()
+
+	wkrQueue, err := queue.NewPgQueue(db.GetUrl())
+	require.NoError(s.T(), err)
+	s.queue = wkrQueue
+
+	wrk := worker.NewTaskWorkerPool(&wkrQueue, m.NewMetrics(prometheus.NewRegistry()))
+	wrk.RegisterHandler(config.DeleteRepositorySnapshotsTask, tasks.DeleteSnapshotHandler)
+	wrk.HeartbeatListener()
+
+	wkrCtx := context.Background()
+	go (wrk).StartWorkers(wkrCtx)
+	go func() {
+		<-wkrCtx.Done()
+		wrk.Stop()
+	}()
+	// Force local storage for integration tests
+	config.Get().Clients.Pulp.StorageType = "local"
```

Comment:
I'm pretty sure that this will get overwritten by whatever is in `config.yaml` if `config.Get()` gets called again in one of the tests.

Assuming this can also be set by env variable, you could do something like `t.SetEnv(key, val)` instead.

Reply:
just tested it, you're right!